cnf-certification-test/certification/certtool/certtool_test.go

```diff
@@ -15,3 +15,170 @@
+			imageID: configuration.ContainerImageIdentifier{
+				Name:       "testID",
+				Repository: "testRepo",
+				Tag:        "testTag",
+				Digest:     "testDigest",
+			},
+		},
+	}
+
+	// Run the test
+	for _, tc := range testCases {
+		result := GetContainerCertificationRequestFunction(tc.imageID)
+		id, err := result()
+		assert.Nil(t, err)
+		myResult := id.(*api.ContainerCatalogEntry)
+		myResult.ID = tc.imageID.Name
+	}
+}
+
+func TestGetOperatorCertificationRequestFunction(t *testing.T) {
+	// Mock out the CertAPIClient
+	defer func() {
+		CertAPIClient = api.CertAPIClient{}
+	}()
+
+	testCases := []struct {
+		opCertFuncRet    bool
+		opCertFuncRetErr error
+	}{
+		{
+			opCertFuncRet:    true,
+			opCertFuncRetErr: nil,
+		},
+		{
+			opCertFuncRet:    false,
+			opCertFuncRetErr: nil,
+		},
+		{
+			opCertFuncRet:    true,
+			opCertFuncRetErr: errors.New("this is an error"),
+		},
+	}
+
+	for _, tc := range testCases {
+		// Mock out the return values for the API func
+		CertAPIClient = &api.CertAPIClientFuncsMock{
+			IsOperatorCertifiedFunc: func(org, packageName, version string) (bool, error) {
+				return tc.opCertFuncRet, tc.opCertFuncRetErr
+			},
+		}
+
+		resultFunc := GetOperatorCertificationRequestFunction("", "", "")
+		certifiedResult, err := resultFunc()
+		if tc.opCertFuncRetErr != nil {
+			assert.NotNil(t, err)
+		} else {
+			assert.Nil(t, err)
+		}
+		testResult := certifiedResult.(bool)
+		assert.Equal(t, tc.opCertFuncRet, testResult)
+	}
+}
+
+func TestCompareVersion(t *testing.T) {
+	// Note: These values pertain to 'kubeVersion' fields found:
+	// https://charts.openshift.io/index.yaml
+	testCases := []struct {
+		ver1           string
+		ver2           string
+		expectedOutput bool
+	}{
+		{
+			ver1:           "1.18.1",
+			ver2:           ">= 1.19",
+			expectedOutput: false,
+		},
+		{
+			ver1:           "1.19.1",
+			ver2:           ">= 1.19",
+			expectedOutput: true,
+		},
+		{
+			ver1:           "1.19",
+			ver2:           ">= 1.16.0 < 1.22.0",
+			expectedOutput: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		assert.Equal(t, tc.expectedOutput, CompareVersion(tc.ver1, tc.ver2))
+	}
+}
+
+func TestWaitForCertificationRequestToSuccess(t *testing.T) {
+	testCases := []struct {
+		testFunc   func() (interface{}, error)
+		expectedID string
+		timeout    time.Duration
+	}{
+		{
+			testFunc: func() (interface{}, error) {
+				return &api.ContainerCatalogEntry{
+					ID: "id1",
+				}, nil
+			},
+			expectedID: "id1",
+			timeout:    time.Second * 5,
+		},
+		{
+			testFunc: func() (interface{}, error) {
+				return &api.ContainerCatalogEntry{
+					ID: "id1",
+				}, nil
+			},
+			expectedID: "id1",
+			timeout:    -1,
+		},
+		{
+			testFunc: func() (interface{}, error) {
+				return &api.ContainerCatalogEntry{
+					ID: "id1",
+				}, errors.New("this is an error")
+			},
+			expectedID: "id1",
+			timeout:    -1,
+		},
+	}
+
+	for _, tc := range testCases {
+		if tc.timeout > time.Second*0 {
```

Comment:
replace with:
		result := WaitForCertificationRequestToSuccess(tc.testFunc, tc.timeout)
		if tc.timeout > time.Second*0 {
			assert.Equal(t, tc.expectedID, result.(*api.ContainerCatalogEntry).ID)
		} else {
			// Note: Cannot cast the result here because the interface returned is nil.
			assert.Nil(t, result)
		}

Reply:
Fixed!